### PR TITLE
Allow NPCs to define cargo

### PR DESCRIPTION
**Pull request by warp-core**
**Feature:** This PR implements the feature request detailed and discussed in issue #7112 

Also, can be used for #4050.

## Feature Details
Moved the `cargo`, `commodities`, and `outfitters` members of `Fleet` to a new class, `FleetCargo`. `Fleet` and `NPC` each have a `FleetCargo` object. Also, moved `Fleet::SetCargo()` to `FleetCargo` along with a bunch of anonymous namespace methods in `Fleet.cpp` used by `SetCargo()`.
If an NPC was given a cargo definition, when it calls `Place()` on a `Fleet`, it will additionally pass a boolean argument telling it not to add cargo to its ships (this is defaulted to true so anything else calling this method should not be affected).
The NPC will instead use its own `FleetCargo` object to add cargo to all of its ships at the end of instantiation.

## Usage Examples
`cargo`, `commodities`, and `outfitters` can be used as child nodes of `npc`, the same way they already could for `fleet`s.

## Testing Done
Very little so far. It built and launched and let me take off without crashing, though.

## Performance Impact
N/A, I hope.

### CODE REVIEW

File: source/FleetCargo.cpp [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+			outfitters.insert(GameData::Outfitters().Get(node.Token(i)));
+	}
+}
+
+
+
+// Choose the cargo associated with this ship.
+// If outfits were specified, but not commodities, do not pick commodities.
+// If commodities were specified, but not outfits, do not pick outfits.
+// If neither or both were specified, choose commodities more often..
```
Comment (tibetiroka) [inline source/FleetCargo.cpp:157]:
```suggestion
// If neither or both were specified, choose commodities more often.
```

**Review by tibetiroka — CHANGES_REQUESTED** [review]
Minor typo fix in a comment.

File: source/FleetCargo.cpp [hunk 2]
```diff
@@ -0,0 +1,196 @@
+/* FleetCargo.cpp
+Copyright (c) 2022 by warp-core
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
```
Comment (tibetiroka) [inline source/FleetCargo.cpp:10]:
```suggestion
PARTICULAR PURPOSE. See the GNU General Public License for more details.

You should have received a copy of the GNU General Public License along with
this program. If not, see <https://www.gnu.org/licenses/>.
```
The standard copyright header was changed recently.

File: source/FleetCargo.h [hunk 3]
```diff
@@ -0,0 +1,35 @@
+/* FleetCargo.h
+Copyright (c) 2022 by warp-core
+
+Endless Sky is free software: you can redistribute it and/or modify it under the
+terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later version.
+
+Endless Sky is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE.  See the GNU General Public License for more details.
```
Comment (tibetiroka) [inline source/FleetCargo.h:10]:
```suggestion
PARTICULAR PURPOSE. See the GNU General Public License for more details.

You should have received a copy of the GNU General Public License along with
this program. If not, see <https://www.gnu.org/licenses/>.
```

File: source/NPC.cpp [hunk 4]
```diff
@@ -113,6 +113,8 @@ void NPC::Load(const DataNode &node)
 			government = GameData::Governments().Get(child.Token(1));
 		else if(child.Token(0) == "personality")
 			personality.Load(child);
+		else if(child.Token(0) == "cargo" || child.Token(0) == "commodities" || child.Token(0) == "outfitters")
```
Comment (Hurleveur) [inline source/NPC.cpp:116]: Isnt this line too long?

**Review by Hurleveur — APPROVED** [review]
Looks very good (most of it is just moving code around, too)

**Comment by warp-core** [comment]
Rereading the summary, I'd actually like to make a change to this. Always overriding cargo defined by the fleet is bad.

**Comment by Hurleveur** [comment]
Well maybe just add a warning that it will be ovewriten?

Some fleets that spawn in a system could have cargo that we'd want to remove or change for a specific mission, without redefining the whole fleet
Also what happens if you define multiple fleets with cargo within one npc; they all get that same cargo, right?
And multiple ships share it as a fleet?

File: source/NPC.cpp (hunk 4 above)
Comment (warp-core) [reply source/NPC.cpp:116]: Over 80, but not quite 120.

File: source/Fleet.h [hunk 5]
```diff
@@ -64,7 +65,7 @@ class Fleet {
 	void Enter(const System &system, std::list<std::shared_ptr<Ship>> &ships, const Planet *planet = nullptr) const;
 	// Place a fleet in the given system, already "in action." If the carried flag is set, only
 	// uncarried ships will be added to the list (as any carriables will be stored in bays).
-	void Place(const System &system, std::list<std::shared_ptr<Ship>> &ships, bool carried = true) const;
+	void Place(const System &system, std::list<std::shared_ptr<Ship>> &ships, bool carried = true, bool addCargo = true) const;
```
Comment (warp-core) [inline source/Fleet.h:68]:
```suggestion
	void Place(const System &system, std::list<std::shared_ptr<Ship>> &ships,
			bool carried = true, bool addCargo = true) const;
```

**Pull request merged**